### DISCUSSION TITLE
Electric-sql for file-first approach

### DISCUSSION THREAD

**Discussion by ctwhome** [electric-sql/electric#1007] [Questions]
hi! I am building a note-taking application with an emphasis on a file-first approach, meaning that the application (web with electron) will make changes in the file, and those need to be real-time in sync with all collaborators and user other devices (mobile, tablet).

Would it be possible to have such a system with electric-sql? the app also needs to handle media files like videos, images, and audio, all of them stored in the computer file system of the user, and somehow syncing everywhere?

**Comment by balegas** [comment]
ElectricSQL is best fit for synchronizing relational data. We've used Electric to store small data files in-database, but for large media files is best to use an external storage. 
It's possible to use ElectricSQL alongside a storage service and associate URLs and their respective data.

**Reply by ctwhome** [reply]
The idea is that all data files and media are stored in the hard disk of the user, then synced everywhere, an approach of how Obsidian.md does, although they do not share their stack for the synchronization part. I guess the whole thing should behave like Google Drive, always listening for file changes, but also keeping indexes in a database-like to perform searches. Could that be possible Electric-sql then?

**Reply by KyleAMathews** [reply]
You could store a hash of files in the db and upload changes to e.g. s3 and then clients download new files as they see them.

You should also checkout electric's experiments with yjs as you could use yjs to sync changes to text files.

**Reply by ctwhome** [reply]
Thanks for the answers, i think it is worth the try to have the ultimate offline mode, file ownership, with realtime sync between active participants... it sounds like a holy grail to me.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
